packages/gatsby-source-filesystem/src/gatsby-node.js

```diff
@@ -7,22 +7,29 @@ exports.sourceNodes = (
   { boundActionCreators, getNode, hasNodeChanged, reporter },
   pluginOptions
 ) => {
-  const { createNode, deleteNode } = boundActionCreators
+  if (!(pluginOptions && pluginOptions.path)) {
+    reporter.panic(`
+You should pass 'path' options for gatsby-source-filesystem plugin
 
-  let ready = false
+See docs here - https://www.gatsbyjs.org/packages/gatsby-source-filesystem/
+    `)
+  }
 
   // Validate that the path exists.
   if (!fs.existsSync(pluginOptions.path)) {
-    console.log(`
+    reporter.panic(`
 The path passed to gatsby-source-filesystem does not exist on your file system:
 
 ${pluginOptions.path}
 
 Please pick a path to an existing directory.
       `)
-    process.exit(1)
   }
 
+  const { createNode, deleteNode } = boundActionCreators
```

Comment:
what are these for?

Reply:
This is existing code, just moved them down after the plugin options validation.

Reply:
👍